Automattic/jetpack
# Instant Search: Add initial framework

#### Changes proposed in this Pull Request:
* Adds the build process and initial code for JP Search as you type front end Preact app
* Loads the front end assets when `JETPACK_SEARCH_PROTOTYPE` is defined as true
* The js code is a very minimal version that really doesn't do anything and will need to be heavily rewritten.

#### Testing instructions:
* Run `yarn build` and the distribution in `_inc/search/dist` should get created
* Add `define( "JETPACK_SEARCH_PROTOTYPE", true );` to your wp-config.php
* In order to load the code you need to have a JP Pro plan and have JP Search enabled.
* Load the front end of your test site and you should see `jp-search.bundle.js` get loaded. The front end site should not be affected otherwise.
* Try also `yarn build-production`. Front end js code should be pretty small.

#### Proposed changelog entry for your changes:
* No changelog for this.
